sopel/tools/__init__.py

```diff
@@ -168,10 +169,31 @@ def decorator(func):
     def deprecated_func(*args, **kwargs):
         if not (warning_in and
                 parse_version(warning_in) >= parse_version(__version__)):
-            stderr(text)
-            # Only display the last frame
+            original_frame = inspect.stack()[-stack_frame]
+            mod = inspect.getmodule(original_frame[0])
+            module_name = None
+            if mod:
+                module_name = mod.__name__
+            if module_name:
+                if 'sopel.' in module_name:
```

Comment:
I didn't realized that first but I think this should be better:

```suggestion
                if module_name.startswith('sopel.'):
```

Reply:
Oh yes, this is fair. It's probably what I intended, lol. Will apply this _and_ tweak the `.replace()` below so it only removes `sopel_modules.` from the package name if that's how it starts.

Reply:
All good!